packages/store/src/patterns/encodePassable.js

```diff
@@ -296,29 +309,64 @@ export const makeEncodeKey = encodeRemotable => {
         );
         return result;
       }
+      case 'error': {
+        const result = encodeError(key);
+        assert(
+          result.startsWith('!'),
+          X`internal: Error encoding must start with "!": ${result}`,
+        );
+        return result;
+      }
+      case 'promise': {
+        const result = encodePromise(key);
+        assert(
+          result.startsWith('?'),
+          X`internal: Promise encoding must start with "p": ${result}`,
+        );
+        return result;
+      }
```

Comment:
Mostly this PR is renaming, but this bit here makes me nervous.  I would like to see tests verifying that even though these encodings now exist, these types still can't be used in contexts where they're not allowed (e.g., in durable data).  Since it was not previously possible to express those disallowed cases I'm not confident that the implementations of persistent stores and virtual objects are prepared to be subjected to these.

Reply:
Check my understanding:

You currently only use `encodeKey` on keys of collections --- the elements of sets and the keys of maps. For the values of maps, you currently use *only* marshal, not encodeKey, correct?

If so, is your enforcement that the key is a `Key` only by calling `encodeKey`, or is there another enforcement point?

Reply:
In any case, even if it is the only enforcement point, your collection manager calls it only with the `encodeRemotable` argument. The other two arguments, `encodeError` and `encodePromise` which the code above calls, default to functions that always throw an error. However, this is way too subtle and I'm happy to make that implicit restriction more explicit.

Reply:
> You currently only use encodeKey on keys of collections --- the elements of sets and the keys of maps. For the values of maps, you currently use only marshal, not encodeKey, correct?

Yes.  I may have garbled things a bit by talking about durable objects in the same sentence.

Reply:
> In any case, even if it is the only enforcement point, your collection manager calls it only with the encodeRemotable argument. The other two arguments, encodeError and encodePromise which the code above calls, default to functions that always throw an error. However, this is way too subtle and I'm happy to make that implicit restriction more explicit

I don't know if you need to do anything besides have a test that tries to use a promise or an error as a key and successfully fails.  Actually, I think such would drop in very nicely as additional cases in `test-durabilityChecks.js`

Reply:
Tests added. PTAL.

Reply:
Nice.